Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -39,10 +39,13 @@ namespace AZ
             Data::AssetHandler::LoadResult loadResult = Data::AssetHandler::LoadResult::Error;
             if (assetData)
             {
-                loadResult = Utils::LoadObjectFromStreamInPlace<StreamingImageAsset>(*stream, *assetData, m_serializeContext) ?
-                    Data::AssetHandler::LoadResult::LoadComplete :
-                    Data::AssetHandler::LoadResult::Error;
-                
+                if (stream->GetLength() > 0)
```

Comment:
Right now, it doesn't happen because of other changes I made, but there's nothing _preventing_ it from happening again. 

The way it happened before was that the AssetUtils::AsyncAssetLoader was passing "true" to auto-generate an ID in the AssetUtils::GetAssetIdFromProductPath method:
https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Asset/AssetUtils.inl#L22
https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Reflect/Asset/AssetUtils.cpp#L39-L44

The AsyncAssetLoader is only used in one place right now, which is here:
https://github.com/o3de/o3de/blob/development/Gems/Atom/Feature/Common/Code/Source/CoreLights/LtcCommon.cpp#L52-L55

If these images weren't compiled by the time this was called, the "true" would cause an entry to get added into the AssetCatalog for an empty 0-length stream for these assets. At that point, the asset load would try to load the asset, see the 0 length, and immediately go into the StreamingImageAssetHandler::LoadAssetData. The 0-length buffer would get passed into the ObjectStream code, and would assert (and maybe crash, I forget, but probably just asserts).

In this PR, I've removed the code that allows you to auto-register an ID with the asset catalog, so instead the assets will show up as missing, which will cause them to successfully load the StreamingImage placeholder assets instead and _in theory_ should cause it to get auto-replaced with the correct assets assuming StreamingImage handles the reloads correctly without LtcCommon needing to do anything (I still need to verify).

Reply:
Oh, and so what I mean by "nothing is _preventing_ it from happening again" is that if anyone else adds code to auto-add entries into the AssetCatalog for a StreamingImageAsset then tries to load it, the 0-length pathway could get triggered again. So it didn't seem like a bad idea to leave in the little safeguard in the loading code.